src/frontend/src/pages/TeamsPage/TeamMembersPageBlock.tsx

```diff
@@ -54,11 +67,16 @@ const TeamMembersPageBlock: React.FC<TeamMembersPageBlockProps> = ({ team }) =>
 
   const hasPerms = auth.user && (isAdmin(auth.user.role) || auth.user.userId === team.head.userId);
 
-  const options = users
+  const memberOptions = users
     .filter((user) => user.userId !== team.head.userId && !team.leads.map((lead) => lead.userId).includes(user.userId))
     .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
     .map(userToAutocompleteOption);
 
+  const headOptions = users
+    .filter((user) => memberOptions.some((option) => option.id === user.userId) && isHead(user.role))
+    .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
+    .map((user) => ({ id: `${userToAutocompleteOption(user).id}`, label: userToAutocompleteOption(user).label }));
```

Comment:
ok this is a problem here (and also everywhere we do this userToAutocompleteOption business), we're using two different autocomplete options in this file. The NERAutoComplete takes an option with an id that's a string, and the AutoComplete is taking an option with an id that's a number. This is confusing. I'd say make userToAutocompleteOption return an option with a string id instead of a number id and then just fix the errors that crop up, and you can return just `usertoAutoCompleteOption(user) ` instead of `{ id: {userToAutocompleteOption(user).id}, label: userToAutocompleteOption(user).label }` throughout this file